### PR TITLE
Implement wlr_box functions

### PR DESCRIPTION
**Pull request by dominikh**

### CODE REVIEW

File: src/types/box.zig [hunk 1]
```diff
@@ -1,8 +1,32 @@
...
+    extern fn wlr_box_intersection(dest: *Box, box_a: *const Box, box_b: *const Box) bool;
+    pub const intersection = wlr_box_intersection;
+
+    extern fn wlr_box_contains_point(box: *const Box, x: f64, y: f64) bool;
+    pub const containsPoint = wlr_box_contains_point;
+
+    extern fn wlr_box_empty(box: *const Box) bool;
+    pub const empty = wlr_box_empty;
+
+    extern fn wlr_box_transform(dest: *Box, box: *const Box, transform: wlr.Output.Transform, width: c_int, height: c_int) void;
```
Comment (ifreund) [inline src/types/box.zig:22]: This enum is from wayland not wlroots. (You'll need an `@include` as well).
```suggestion
    extern fn wlr_box_transform(dest: *Box, box: *const Box, transform: wl,Output.Transform, width: c_int, height: c_int) void;
```

File: src/types/box.zig [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+    pub const transform = wlr_box_transform;
+
+    extern fn wlr_box_rotated_bounds(dest: *Box, box: *const Box, rotation: f32) void;
+    pub const rotatedBounds = wlr_box_rotated_bounds;
+
+    extern fn wlr_box_from_pixman_box32(dest: *Box, box: *const pixman.Box32) void;
```
Comment (ifreund) [inline src/types/box.zig:28]: This looks to be passed by value. Why that is the case I do not know. 
```suggestion
    extern fn wlr_box_from_pixman_box32(dest: *Box, box: pixman.Box32) void;
```

**Comment by dominikh** [comment]
PTAL

**Review by ifreund — APPROVED** [review]
LGTM, thanks!

**Pull request merged**